irqstat

```diff
@@ -44,25 +44,49 @@ import threading
 KEYEVENT = threading.Event()
 
 
-def gen_numa():
+def gen_numa(numafile):
     """Generate NUMA info"""
     cpunodes = {}
     numacores = {}
-    out = subprocess.Popen('numactl --hardware | grep cpus', shell=True,
-                           stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    errtxt = out.stderr.readline()
-    if errtxt:
-        print errtxt + '\r\n'
-        print "Is numactl installed?\r"
+    s = ""
+
+    try:
+        if not numafile:
+            temp = subprocess.Popen(['numactl', '--hardware'],
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+            (output, error) = temp.communicate()
+            status = temp.wait()
+
+            if error:
+                print "NUMACTL ERROR:"
+                print(error)
+                exit(1)
+        else:
+            nf = open(numafile, 'r')
+            output = nf.read()
+            nf.close()
+
+        output = output.split("\n")
+        for line in output:
```

Comment:
Much better than using grep, thanks.

Reply:
Yeah, well, I'm using this as an opportunity to finally try to learn python properly :-)